messenger.go

```diff
@@ -916,6 +935,111 @@ func (p *postProcessor) matchMessage(message *protocol.Message, chats []*Chat) (
 	}
 }
 
+type EnsDetails struct {
```

Comment:
1. IMO it should be a separate file at least. Like `ens.go`.
2. Small detail but `ENSDetails` and `ENSRespoonse` is a recommended way:
```
Words in names that are initialisms or acronyms (e.g. "URL" or "NATO") have a consistent case. For example, "URL" should appear as "URL" or "url" (as in "urlPony", or "URLPony"), never as "Url". Here's an example: ServeHTTP not ServeHttp.

This rule also applies to "ID" when it is short for "identifier," so write "appID" instead of "appId".
```